doc/developer_guide/transformations.rst

```diff
@@ -516,4 +516,71 @@ an if condition or a loop condition, and that intrinsic contains an array sectio
+
+
+This update is happening gradually, with developers being asked to update
+transformations as they are otherwise being modified.
+
+Note that the while the `options` dict is being deprecated, it is still
+accepted and ovverides the keyword arguments if both are provided.
+
+The steps required to update the transformations are detailed here (see
+the `ParallelLoopTrans` class for reference):
+
+1. Update the `apply` function to have a keyword argument for each available
+   option. These arguments should have type hints and default values defined
+   if possible. The `options` argument should remain, and `**kwargs` should
+   also be available. If type hints are not possible for one of the keyword
+   arguments, then type hints will need to be provided in the documentation
+   as normal, and type checking will need to be done manually in validate as
+   `validate_options` cannot automatically do type checking where no type hint
+   is available. Note that if there is no `apply` function it should be added
+   if the Transformation takes any options as the new infrastructure is
+   built upon the `apply` definition (e.g. `LoopTrans` has
+   validation used for subclasses, but performs no actions in its newly added
+   `apply` routine).
+2. The `validate` function should call the `validate_options` routine on each of
+   the keyword arguments and `**kwargs`. This routine should not be called on
+   the `options` dictionary. The `options` input should overrule the keyword
+   arguments when determining options to the apply and validate function.
+   There should also be a `# TODO 2668` to remove the options argument later:
+   
+.. code-block:: python
+
+    if not options:
+        self.validate_options(tiling=tiling, **kwargs)
+    else:
+        #TODO #2668: Deprecate options dictionary.
+        tiling = options.get("tiling", False)
+
+3. Modify the `validate` function to take `**kwargs` (but no other keyword
+   arguments other than options). Use the `get_option('optname', **kwargs)`
+   to get each of the options out of the kwargs, as this will also give
+   the default value if the option isn't passed into validate.
+4. If the `apply` function doesn't call `super().apply` all the way up to
+   the base `Transformation` class, then it should print the
+   `self._deprecation_warning` message if the `options` dict is provided.
+5. Rewrite the documentation on the `apply` function to document the new
+   keyword arguments, and remove any reference to options. Remove the
+   documentation on the keyword arguments to `validate`, as these will be
+   generated automatically by PSyclone.
+6. Import the `transformation_documentation_wrapper` from `psyclone.utils` and
+   add `@transformation_documentation_wrapper` to the class definition. If
+   there are no calls to super.apply() or super.validate(), then disable
+   option inheritance in the documentation by using
+   `@transformation_documentation_wrapper(inherit=False)`.
```

Comment:
I think I would bring this to number 1, as this is the first step to start testing things incrementally, otherwise nothing works.

Reply:
Done.